src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -401,6 +409,115 @@ static void PromoteCommonCommitCharactersOntoList(LSP.VSCompletionList completio
+            CancellationToken cancellationToken)
+        {
+            var position = await document.GetPositionFromLinePositionAsync(ProtocolConversions.PositionToLinePosition(request.Position), cancellationToken).ConfigureAwait(false);
+            var completionTrigger = await ProtocolConversions.LSPToRoslynCompletionTriggerAsync(request.Context, document, position, cancellationToken).ConfigureAwait(false);
+
+            // Get the cached completion list in the case of a request for an incomplete list or calculate a new list and cache it.
+            CompletionList completionList;
+            long resultId;
+            if (request.Context?.TriggerKind == LSP.CompletionTriggerKind.TriggerForIncompleteCompletions)
+            {
+                // This is a request for a list we already calculated but was incomplete.
+                // Use the resultId to retrieve the completion list we cached for the original result.
+                Contract.ThrowIfFalse(_lastIncompleteResultId.HasValue, "Trigger for incomplete list missing previous result id");
+                var cachedList = _completionListCache.GetCachedCompletionList(_lastIncompleteResultId.Value);
+                Contract.ThrowIfNull(cachedList, "Trigger for incomplete list missing cached list");
+
+                completionList = cachedList.CompletionList;
+                resultId = _lastIncompleteResultId.Value;
+            }
+            else
+            {
+                // This is a new completion request, clear out the last result Id for incomplete results.
+                _lastIncompleteResultId = null;
+
+                completionList = await completionService.GetCompletionsAsync(document, position, completionTrigger, options: completionOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
+                if (completionList == null || completionList.Items.IsEmpty || cancellationToken.IsCancellationRequested)
+                {
+                    return null;
+                }
+
+                // Cache the completion list so we can avoid recomputation in the resolve handler
+                resultId = _completionListCache.UpdateCache(request.TextDocument, completionList);
+            }
+
+            // Determine the current filter text based on the position in the document.
+            var filterText = sourceText.GetSubText(completionService.GetDefaultCompletionListSpan(sourceText, position)).ToString();
+
+            // Use pattern matching to determine which items are most relevant out of the calculated items.
+            using var _ = ArrayBuilder<MatchResult<CompletionItem?>>.GetInstance(out var matchResultsBuilder);
+            var index = 0;
+            var completionHelper = CompletionHelper.GetHelper(document);
+            foreach (var item in completionList.Items)
+            {
+                if (CompletionHelper.TryCreateMatchResult<CompletionItem?>(
+                    completionHelper,
```

Comment:
please indent all the arguments from tneh caller level.

Reply:
This was the result of the wrap refactoring
![image](https://user-images.githubusercontent.com/5749229/120378553-6dbe2d00-c2d3-11eb-9083-6c06d808381e.png)

I'm not sure if this is wrong, are you expecting an indentation starting at the column where CompletionHelper starts, instead of where the if starts?

Reply:
darn wrap refactoring!  who wrote that POS. :-/

> are you expecting an indentation starting at the column where CompletionHelper starts, instead of where the if starts?

personally yes.  i dont' like children of a node starting at the same idnentation as their parent.  it makes them appear as siblings to me.  

(i don't believe our formatting system does well in cases like this though :-/ )